ArdentHQ/arkscan
# refactor: handle repeated livewire request exceptions

## Summary

https://app.clickup.com/t/86dqmbkz9

Created a JS handler to watch for failed livewire requests. When the failure count reaches 10 failed requests in a row (per component), the page will reload which will force the component to reload. There are a few possible reasons a livewire request could fail:

- an error with the backend of the component
- there is livewire corruption (e.g. the backend has properties which the frontend did not) - common with new releases
- lost internet connection?

10 repeatedly failed attempts to update a component seemed like a safe number. The Price Ticker is updated every 30 seconds so it would take ~5 minutes for this to trigger. However, the delegate monitor polls every second (ish) so that could take 10-20 seconds before triggering a page reload.

To test:

- Open `resources/views/livewire/price-ticker.blade.php`
- Change the polling from `30s` to `1s`
- Load any page of your local explorer in your browser **and leave it open**
- Open `app/Http/Livewire/PriceTicker.php`
- Add `public string $price;` to the properties
- Add `$this->price = md5((string) now()->unix());` to the mount method
- Save
- Should get the exception in the window you opened
- Wait for 10 failed requests
- Page should reload







## Checklist
